src/networks/evm/client.ts

```diff
@@ -206,42 +279,38 @@ export class EthereumClient extends Client {
       data: encodedTx,
     };
 
-    const gas = await this.web3.eth.estimateGas(txObject);
-    const gasPrice = Number(await this.web3.eth.getGasPrice());
-    const nonce = await this.web3.eth.getTransactionCount(address);
-    txObject.gas = gas;
-    txObject.gasPrice = `0x${BigInt(Math.ceil(gasPrice * this.gasMultiplier)).toString(16)}`;
+    const txFee = await this.estimateTxFee(txObject);
+    const nonce = await this.getNativeNonce();
+    txObject.gas = txFee.gas.toString();
+    txObject.gasPrice = `0x${BigInt(Math.ceil(Number(txFee.gasPrice) * this.gasMultiplier)).toString(16)}`;
```

Comment:
And here :)

Reply:
https://github.com/zkBob/zeropool-support-js/pull/9/commits/96c6edc93af0da09efec3308f6df99d4df414046